Mathlib/Analysis/Analytic/Composition.lean

```diff
@@ -860,17 +860,32 @@ alias AnalyticWithinOn.comp := AnalyticOn.comp
 
 /-- If two functions `g` and `f` are analytic respectively at `f x` and `x`, then `g ∘ f` is
 analytic at `x`. -/
+@[fun_prop]
 theorem AnalyticAt.comp {g : F → G} {f : E → F} {x : E} (hg : AnalyticAt 𝕜 g (f x))
     (hf : AnalyticAt 𝕜 f x) : AnalyticAt 𝕜 (g ∘ f) x := by
   rw [← analyticWithinAt_univ] at hg hf ⊢
   apply hg.comp hf (by simp)
 
+/-- If two functions `g` and `f` are analytic respectively at `f x` and `x`, then `g ∘ f` is
+analytic at `x`. -/
+@[fun_prop]
+theorem AnalyticAt.comp' {g : F → G} {f : E → F} {x : E} (hg : AnalyticAt 𝕜 g (f x))
+    (hf : AnalyticAt 𝕜 f x) : AnalyticAt 𝕜 (fun z ↦ g (f z)) x := by
+  apply hg.comp hf
+
 /-- Version of `AnalyticAt.comp` where point equality is a separate hypothesis. -/
+@[fun_prop]
```

Comment:
```suggestion
```
This one shouldn't be marked, because `fun_prop` will never be able to provide the auxiliary hypothesis unless it is trivial. In this case, the versions above will apply.

Reply:
Roger. Thanks for pointing that out.